src/main/java/uk/gov/companieshouse/appointments/subdelta/logging/LoggingKafkaListenerAspect.java

```
@@ -55,7 +77,7 @@ public Object manageStructuredLogging(ProceedingJoinPoint joinPoint)
 
     private Optional<String> extractContextId(Object payload) {
         if (payload instanceof ResourceChangedData) {
-            return Optional.of(((ResourceChangedData)payload).getContextId());
+            return Optional.of(((ResourceChangedData) payload).getContextId());
```

We should throw an exception if the payload is the wrong type instead of returning an optional
e.g. from filing history
```
    private ChsDelta extractChsDelta(Object payload) {
        if (payload instanceof ChsDelta chsDelta) {
            return chsDelta;
        }
        throw new NonRetryableException("Invalid payload type. payload: %s".formatted(payload.toString()));
    }
```